contracts/UniswapV2Router01.sol

```diff
@@ -133,14 +111,16 @@ contract UniswapV2Router01 is IUniswapV2Router01, NilCurrencyBase {
         return UniswapV2Library.getAmountOut(amountOut, reserveIn, reserveOut);
     }
 
-    function getAmountsOut(uint amountIn, address[] memory path) public view override returns (uint[] memory amounts) {
-        return UniswapV2Library.getAmountsOut(factory, amountIn, path);
-    }
-
-    function getAmountsIn(uint amountOut, address[] memory path) public view override returns (uint[] memory amounts) {
-        return UniswapV2Library.getAmountsIn(factory, amountOut, path);
+    receive() external payable {
     }
 
-    receive() external payable {
+    function smartCall(address dst, Nil.Token[] memory tokens, bytes memory callData) private returns (bool) {
+        if (Nil.getShardId(dst) == Nil.getShardId(address(dst))) {
```

Comment:
Not sure i understand this condition, shouldn't we compare with address(this)

Reply:
you're righ, fixed